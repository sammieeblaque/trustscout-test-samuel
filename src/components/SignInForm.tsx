import { useState } from "react";
import { Button, Form, Card, Alert } from "react-bootstrap";
import {useHistory } from 'react-router-dom'
import { SignUpLink } from "./SignupForm";

import * as routes from "../routes";
import { auth } from "../firebase";



const SignInForm = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState<Object>({});
  const [showAlert, setShowAlert] = useState(false);
  const history = useHistory()
  
   const onSubmit = async (e: any) => {
    e.preventDefault();
    try {
      auth.doSignInWithEmailAndPassword(email, password);
      Promise.all([setEmail(""), setPassword("")])
      history.push(routes.HOME)
    } catch (error) {
      Promise.all([setError(error),setShowAlert(true)])
    }
  };

  return (
    <>
       <div data-testid="signin">
          <Card className="no__border left__pad">
            <Card.Body className="w__90">
              <h2 className="text-center mb-4">Sign In</h2>
            {showAlert && (
              <Alert
                variant="danger"
                onClose={() => setShowAlert(false)}
                dismissible
              >
                { error?.message}
              </Alert>
            )}
            <Form onSubmit={onSubmit}>
              <Form.Group id="email">
                <Form.Label>Email</Form.Label>
                <Form.Control
                  placeholder="john@xyz.com"
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                  required
                />
              </Form.Group>
              <Form.Group id="password">
                <Form.Label>Password</Form.Label>
                <Form.Control
                  type="password"
                  value={password}
                  onChange={(e) =>
                    setPassword(e.target.value)
                  }
                  required
                />
              </Form.Group>
              <Button className="w-100 button__color" type="submit">
                Sign In
              </Button>
              <div className="text-center">
                <SignUpLink />
              </div>
            </Form>
          </Card.Body>
        </Card>
      </div>
    </>
  )
}

export { SignInForm };
