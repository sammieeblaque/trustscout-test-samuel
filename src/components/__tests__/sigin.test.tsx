import { render, cleanup, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { SignInForm } from "../SignInForm";

afterEach(() => {
  cleanup();
});

describe("to run tests", () => {
  test("should render SignIn", () => {
    let history;
    render(<SignInForm history={history} />);
    const signinElement = screen.getByTestId("signin");
    expect(signinElement).toBeInTheDocument();
  });
});

test("Should match snapshot", () => {
  let history;
  const tree = renderer.create(<SignInForm history={history} />).toJSON();
  expect(tree).toMatchSnapshot();
});
