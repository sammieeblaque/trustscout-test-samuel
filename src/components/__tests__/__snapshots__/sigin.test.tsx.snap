// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  data-testid="signin"
>
  <div
    className="no__border left__pad card"
  >
    <div
      className="w__90 card-body"
    >
      <h2
        className="text-center mb-4"
      >
        Sign In
      </h2>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
          id="email"
        >
          <label
            className="form-label"
          >
            Email
          </label>
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="john@xyz.com"
            required={true}
            value=""
          />
        </div>
        <div
          className="form-group"
          id="password"
        >
          <label
            className="form-label"
          >
            Password
          </label>
          <input
            className="form-control"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
        <button
          className="w-100 button__color btn btn-primary"
          disabled={false}
          type="submit"
        >
          Sign In
        </button>
        <div
          className="text-center"
        />
      </form>
    </div>
  </div>
</div>
`;

exports[`Should match snapshot 1`] = `
<div
  data-testid="signin"
>
  <div
    className="no__border left__pad card"
  >
    <div
      className="w__90 card-body"
    >
      <h2
        className="text-center mb-4"
      >
        Sign In
      </h2>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
          id="email"
        >
          <label
            className="form-label"
          >
            Email
          </label>
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="john@xyz.com"
            required={true}
            value=""
          />
        </div>
        <div
          className="form-group"
          id="password"
        >
          <label
            className="form-label"
          >
            Password
          </label>
          <input
            className="form-control"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
        <button
          className="w-100 button__color btn btn-primary"
          disabled={false}
          type="submit"
        >
          Sign In
        </button>
        <div
          className="text-center"
        >
          <p>
            Don't have an account? 
            <a
              href="/signup"
            >
              Sign Up
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</div>
`;
